compiler/noirc_frontend/src/hir/comptime/value.rs

```diff
@@ -345,7 +348,8 @@ impl Value {
                 HirExpression::Literal(HirLiteral::Slice(HirArrayLiteral::Standard(elements)))
             }
             Value::Quoted(tokens) => HirExpression::Unquote(add_token_spans(tokens, location.span)),
-            Value::Pointer(..)
+            Value::Expr(..)
```

Comment:
We can convert an expression into a HIR expression by elaborating it.

I honestly need to review our use of `into_hir_expression` though on whether we should keep it at all. It feels wrong to have both it and `into_expression` and the `Value::Quoted` is different with the HIR version returning an `Unquote` expression instead of immediately parsing & elaborating the tokens into their actual hir expr variant.

Reply:
I looked at the code and it seems we don't have an `Elaborator` at this point. Do you want me to try to do that, or we first solve the thing that feels wrong? (not sure what changes I need to make here, if any)

Oh, and to clarify, I should have put a `todo!()` here.

Reply:
My bad, forgot we weren't in the interpreter. Let's leave it as-is for now with the error and we can revisit how into_hir_expression is used later.